jerry-core/ecma/builtin-objects/ecma-builtin-array-prototype.c

```diff
@@ -453,6 +453,18 @@ ecma_builtin_array_prototype_object_pop (ecma_object_t *obj_p, /**< array object
     return get_value;
   }
 
+  if (ecma_get_object_type (obj_p) == ECMA_OBJECT_TYPE_ARRAY
+      && ((ecma_extended_object_t *) obj_p)->u.array.is_fast_mode)
```

Comment:
Could we also add special handling for retrieving the last element instead of using `ecma_op_object_get`?